### ISSUE TITLE
Why duplicating the log_assignment calculation?

### ISSUE THREAD

**Issue by 013292** [cvg/glue-factory#65]
First of all, appreciate for sharing the training code of the light glue.

At this moment, I'm replicating the first round of training on homograph dataset, here is the loss of the lightglue: 

![image](https://github.com/cvg/glue-factory/assets/72900674/5506d7d3-22dd-4db7-89a9-9b149886fec5)
The `loss_params` calculates the `log_assignment` which looks like the same as in the output of lightglue,  why do we need to do that again since there is already a `log_assignment` item in `data`?

**Comment by Phil26AT** [comment]
You are right, this could be reused. We computed it again for safety, it's likely not a big bottleneck during training (for LightGlue you anyway have to recompute it L times, L being the number of attention layers).
